source/Tests/Parsing/AttributeParsingFixture.cs

```diff
@@ -9,11 +9,25 @@ namespace Tests.Parsing
 {
     public class AttributeParsingFixture
     {
-        [Test]
-        public void String()
-            => OclParser.Execute(@"Foo = ""Bar""")
+        [TestCase("Bar", "Bar")]
+        [TestCase(@"\r", "\r")]
+        [TestCase(@"\n", "\n")]
+        [TestCase(@"\t", "\t")]
+        [TestCase(@"\\", @"\")]
+        [TestCase(@"\""", @"""")]
+        [TestCase(@"\\\\\\\""\""", @"\\\""""")]
+
+        public void String(string text, string expected)
+            => OclParser.Execute($@"Foo = ""{text}""")
                 .Should()
-                .HaveChildrenExactly(new OclAttribute("Foo", "Bar"));
+                .HaveChildrenExactly(new OclAttribute("Foo", expected));
+
+        [Test]
+        public void StringInvalidEscape()
+        {
+            Action action = () => OclParser.Execute(@"Foo = ""\q""");
+            action.Should().Throw<OclException>().WithMessage(@"Unrecognised character escape: \q");
```

Comment:
With a bit of fiddling with the parser [here](https://github.com/OctopusDeploy/Ocl/pull/17/files#diff-cf5159c436a4410e0a35fade3eb69f97R10) and maybe a few other places I think we could surface a line number/column here.

Reply:
I'm open to changing the parser alltogether